base_layer/common_types/src/tari_address/single_address.rs

```diff
@@ -0,0 +1,378 @@
+// Copyright 2020. The Tari Project
+//
+// Redistribution and use in source and binary forms, with or without modification, are permitted provided that the
+// following conditions are met:
+//
+// 1. Redistributions of source code must retain the above copyright notice, this list of conditions and the following
+// disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the
+// following disclaimer in the documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse or promote
+// products derived from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+// INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
+// USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+use std::convert::TryFrom;
+
+use serde::{Deserialize, Serialize};
+use tari_common::configuration::Network;
+use tari_crypto::tari_utilities::ByteArray;
+use tari_utilities::hex::{from_hex, Hex};
+
+use crate::{
+    dammsum::{compute_checksum, validate_checksum},
+    emoji::{EMOJI, REVERSE_EMOJI},
+    tari_address::{TariAddressError, TariAddressFeatures, INTERNAL_SINGLE_SIZE},
+    types::PublicKey,
+};
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, Default)]
+pub struct SingleAddress {
+    network: Network,
+    features: TariAddressFeatures,
+    public_spend_key: PublicKey,
+}
+
+impl SingleAddress {
+    /// Creates a new Tari Address from the provided public keys, network and features
+    pub fn new(spend_key: PublicKey, network: Network, features: TariAddressFeatures) -> SingleAddress {
+        Self {
+            network,
+            features,
+            public_spend_key: spend_key,
+        }
+    }
```

Comment:
Specifying the features should not be allowed, as a single address cannot receive one-sided payments. Default for single address must mean INTERACTIVE.
```suggestion
    pub fn new(spend_key: PublicKey, network: Network) -> SingleAddress {
        Self {
            network,
            features: TariAddressFeatures::create_interactive_only(),
            public_spend_key: spend_key,
        }
    }
```

Reply:
I dont agree here with this change, Its a constructed, it should be allowed to change how we want to be.
But agree default for single should be made interactive

Reply:
The current use case only works for `INTERACTIVE`, while `ONE_SIDED` is undefined.